pdudaemon/drivers/tplink.py

```diff
@@ -93,21 +102,62 @@ def send_command(self, json_string):
             log.error(f"Could not connect to host {self.hostname}:9999")
 
     def get_context(self, port_number):
-        for child in self.childinfo:
-            child_port = (int(child['alias'].split("_")[-1]) + 1)
-            if int(port_number) == int(child_port):
-                return ({"child_ids": [child["id"]]})
+        for i, child in enumerate(self.childinfo, start=1):
+            try:
+                port_id = str(child['id'])
+                log.debug(f"child ID is {port_id}")
+
+                # If the port_id contains the device id,
+                if port_id.find(self.device_id) == 0:
+                    # Get the substring that comes after the device_id
+                    # in the port_id. This corresponds to the outlet number
+                    child_port = int(port_id[len(self.device_id)-1:])
+
+                    # Add 1 because outlet numbers are 0 based
+                    child_port = child_port + 1
+
+                if port_number == int(child_port):
+                    return ({"child_ids": [port_id]})
+            except:
+                # log.warning("Oulet device/outlet id issue")
```

Comment:
Bit of leftover debug here?

Reply:
Removed that line of commented code

Reply:
Can you squash the removal of left-over debug into the patch introducing it otherwise it makes patch-by-patch review impossible